cli/cmd/sync_v1.go

```diff
@@ -7,26 +7,46 @@ import (
 	"time"
 
 	"github.com/cloudquery/plugin-sdk/clients/source/v1"
+	pluginsSource "github.com/cloudquery/plugin-sdk/plugins/source"
 	"github.com/cloudquery/plugin-sdk/specs"
 	"github.com/rs/zerolog/log"
 	"github.com/schollz/progressbar/v3"
 	"golang.org/x/sync/errgroup"
 )
 
 func syncConnectionV1(ctx context.Context, cqDir string, sourceSpec specs.Source, destinationsSpecs []specs.Destination, uid string, noMigrate bool) error {
+	var metrics *pluginsSource.Metrics
+	exitReason := "unknown"
+	canceled := false
+	defer func() {
+		// Send analytics, if activated.
+		if analyticsClient != nil {
+			log.Info().Msg("Sending sync summary to " + analyticsClient.Host())
+			if err := analyticsClient.SendSyncMetrics(context.Background(), sourceSpec, destinationsSpecs, uid, metrics, exitReason); err != nil {
+				log.Warn().Err(err).Msg("Failed to send sync summary")
+			}
+		}
+	}()
 	opts := []source.ClientOption{
 		source.WithLogger(log.Logger),
 		source.WithDirectory(cqDir),
 	}
 	if disableSentry {
 		opts = append(opts, source.WithNoSentry())
 	}
-	sourceClient, err := source.NewClient(ctx, sourceSpec.Registry, sourceSpec.Path, sourceSpec.Version, opts...)
+	sourceCtx, sourceCancel := context.WithCancel(context.Background())
```

Comment:
why background here?

Reply:
The idea here is that the CLI should manage the context for the plugins. If the CLI receives a signal and its context gets cancelled, we don't want that to bubble down to the plugins, because then we don't have time to fetch metrics first and send a shutdown signal of our own. Instead, the CLI should listen for when its context gets cancelled, do some clean-up, and then cancel the contexts of the plugins

Reply:
To add to this: now that https://github.com/cloudquery/plugin-sdk/pull/664 has been merged, the plugins should run in a separate process group, so they won't receive an interrupt signal from the OS when ctrl+c is pressed. However, we still use a [command with context](https://github.com/cloudquery/plugin-sdk/blob/main/clients/source/v1/source.go#L129) which [sends a kill signal](https://pkg.go.dev/os/exec#CommandContext) as soon as the context is cancelled. To avoid this, we will either need to stop using CommandContext, or send a different context like the change in this PR is doing.